### PR TITLE
Closes #785: Separate gradient background view into two views

### LINKED ISSUE #785: Focus is missing URL bar when loading first page
I see this regularly on my older GS5 when I load the first page:
screenshot_2017-06-15-13-41-15

It shortly (5 seconds max?) transitions to show the URL bar and the web view. It's ugly but still functional. I don't see this on faster devices.

### PR DESCRIPTION
**Pull request by mcomella**
#785

Tested on:
- N5 (Android 6)
- GS5 (Android 6)

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java [hunk 1]
```diff
@@ -774,8 +794,22 @@ public void setBlockingEnabled(boolean enabled) {
...
+        appBarLayout.setBackgroundResource(backgroundResource);
+        backgroundTransitionGroup = getUpdatedBackgroundTransitionGroup(); // must be called after resource update.
+    }
+
+    /**
+     * Returns an updated drawable group for the background views, based on their current view resources.
+     *
+     * For more details on these background views, see comments around their declaration in fragment_browser.xml.
+     */
+    private TransitionDrawableGroup getUpdatedBackgroundTransitionGroup() {
```
Comment (liuche) [inline app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java]: Nit: I might call this getCurrentBackgroundTransitionGroup - "updated" from what?

File: app/src/main/res/layout/fragment_browser.xml [hunk 2]
```diff
@@ -13,10 +13,20 @@
         android:layout_width="match_parent"
         android:layout_height="match_parent">
 
+        <!-- The toolbar gradient background extends from under the system status bar to the app's Toolbar.
```
Comment (liuche) [inline app/src/main/res/layout/fragment_browser.xml]: nit: through the app's Toolbar

**Comment by liuche** [comment]
Looks like there are some conflicts, so I'll just let mcomella fix and merge on Thurs.

**Comment by pocmo** [comment]
This looks broken on my device (Pixel XL, Android 8.0):

statusbar-broken

**Comment by mcomella** [comment]
> This looks broken on my device (Pixel XL, Android 8.0):

As per Chenxia's comment above, I think what's happening is the returned status bar height is 0 on this device – I'm not sure why but Chenxia pointed out others had this problem in the StackOverflow post where I found the parts of my solution.

The reason the status bar background is visible at all (and does not have a height of 0) is because, I think, the status bar background view is taking the height of the progress bar, which was added to to it so that the two background views are continuous.

**Comment by pocmo** [comment]
I think in some of the layouts we just naively assumed a fixed status bar height of 25?

I'm going to close this PR - just so that it's not in the list indefinitely. However it's still going to be linked from the issue to pick it up again at a later point.

**Pull request closed without merging**